src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -687,8 +689,30 @@ DistributedInsertSelectSupported(Query *queryTree, RangeTblEntry *insertRte,
 							 NULL, NULL);
 	}
 
-	/* we don't support LIMIT, OFFSET and WINDOW functions */
-	DeferredErrorMessage *error = MultiTaskRouterSelectQuerySupported(subquery);
+
+	/*
```

Comment:
Well, I'm not sure if we really need a special case for `LIMIT` here. `DeferErrorIfUnsupportedSubqueryPushdown` already has the check. 

I know that there is a special case regarding LIMIT when SubqueryPushdown is enabled. I'm fine obeying that.

SubqueryPushdown is heavily discouraged anyway, if you try to set, you get the following Notice:
```
NOTICE:  Setting citus.subquery_pushdown flag is discouraged becuase it forces the planner to pushdown certain queries, skipping relevant correctness checks.
DETAIL:  When enabled, the planner skips many correctness checks for subqueries and pushes down the queries to shards as-is. It means that the queries are likely to return wrong results unless the user is absolutely sure that pushing down the subquery is safe. This GUC is maintained only for backward compatibility, no new users are supposed to use it. The planner is capable of pushing down as much computation as possible to the shards depending on the query.
SET
```

All and all, I don't care about `SubqueryPushdown`, and I don't prefer to carry any further special rules around that.

Reply:
You are right that we donot need limit clause check here. `SubqueryPushdown ` is false by default, which makes subqueries with `LIMIT` clause not pushdownable. When a colocated lateral query with limit clause references an outer query, it is still safe to push down `insert select`. I just added 2 tests to show that.